src/providers/SolanaWalletProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import { WalletAdapterNetwork } from "@solana/wallet-adapter-base"
+import {
+  ConnectionProvider,
+  WalletProvider,
+} from "@solana/wallet-adapter-react"
+import { UnsafeBurnerWalletAdapter } from "@solana/wallet-adapter-wallets"
+import { clusterApiUrl } from "@solana/web3.js"
+import React, { type FC, type ReactNode, useMemo } from "react"
+
+// Default styles that can be overridden by your app
+import "@solana/wallet-adapter-react-ui/styles.css"
+import { WalletModalProvider } from "@solana/wallet-adapter-react-ui"
+
+export const SolanaWalletProvider: FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  // The network can be set to 'devnet', 'testnet', or 'mainnet-beta'.
+  const network = WalletAdapterNetwork.Mainnet
+
+  // You can also provide a custom RPC endpoint.
+  const endpoint = useMemo(() => clusterApiUrl(network), [network])
```

Comment:
Public Solana RPC node will be used, won't it rate limit us?

Reply:
Yes, that's right, I haven't found the info yet but I think it will not cause an issue now. Later, in case of such a problem, we will be able to switch to a private one.

Reply:
I asked because when I was playing locally I was banned, but didn't dive into it, just mentioning that it might be a problem.